File: frontend/src/app/core/services/chat.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiResponse } from './auth.service';

export interface ChatRequest {
  question: string;
  documentId?: string;
  sessionId?: string;
}

export interface SourceChunk {
  chunkId: string;
  documentId: string;
  content: string;
  score: number;
}

export interface ChatResponse {
  sessionId: string;
  answer: string;
  sources: SourceChunk[];
}

export interface ChatSessionResponse {
  id: string;
  documentId: string;
  title: string;
  messageCount: number;
  createdAt: string;
}

export interface ChatMessageResponse {
  id: string;
  role: 'USER' | 'ASSISTANT';
  content: string;
  sourceChunks: SourceChunk[];
  createdAt: string;
}

@Injectable({ providedIn: 'root' })
export class ChatService {
  constructor(private http: HttpClient) {}

  chat(request: ChatRequest): Observable<ApiResponse<ChatResponse>> {
    return this.http.post<ApiResponse<ChatResponse>>('/api/ai/chat', request);
  }

  getSessions(): Observable<ApiResponse<ChatSessionResponse[]>> {
    return this.http.get<ApiResponse<ChatSessionResponse[]>>('/api/ai/sessions');
  }

  getSession(id: string): Observable<ApiResponse<ChatMessageResponse[]>> {
    return this.http.get<ApiResponse<ChatMessageResponse[]>>(`/api/ai/sessions/${id}`);
  }

  deleteSession(id: string): Observable<ApiResponse<void>> {
    return this.http.delete<ApiResponse<void>>(`/api/ai/sessions/${id}`);
  }
}

File: backend/ai-service/src/main/java/com/docassist/ai/service/ChatService.java
package com.docassist.ai.service;

import com.docassist.ai.dto.*;
import com.docassist.ai.entity.*;
import com.docassist.ai.repository.ChatMessageRepository;
import com.docassist.ai.repository.ChatSessionRepository;
import com.docassist.common.exception.ResourceNotFoundException;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.UUID;

@Service
@RequiredArgsConstructor
@Slf4j
public class ChatService {

    private final ChatSessionRepository sessionRepository;
    private final ChatMessageRepository messageRepository;
    private final RagService ragService;

    @Transactional
    public ChatResponse chat(ChatRequest request, UUID userId) {
        ChatSession session;

        if (request.getSessionId() != null) {
            session = sessionRepository.findByIdAndUserId(request.getSessionId(), userId)
                    .orElseThrow(() -> new ResourceNotFoundException("ChatSession", "id", request.getSessionId()));
        } else {
            session = ChatSession.builder()
                    .userId(userId)
                    .documentId(request.getDocumentId())
                    .title(request.getQuestion().substring(0, Math.min(100, request.getQuestion().length())))
                    .build();
            session = sessionRepository.save(session);
        }

        ChatMessage userMessage = ChatMessage.builder()
                .session(session)
                .role(MessageRole.USER)
                .content(request.getQuestion())
                .build();
        messageRepository.save(userMessage);

        RagService.RagResult result = ragService.answerQuestion(
                request.getQuestion(), userId, request.getDocumentId());

        ChatMessage assistantMessage = ChatMessage.builder()
                .session(session)
                .role(MessageRole.ASSISTANT)
                .content(result.answer())
                .sourceChunks(result.sources())
                .build();
        messageRepository.save(assistantMessage);

        return ChatResponse.builder()
                .sessionId(session.getId())
                .answer(result.answer())
                .sources(result.sources())
                .build();
    }

    public List<ChatSessionResponse> getUserSessions(UUID userId) {
        return sessionRepository.findByUserIdOrderByCreatedAtDesc(userId)
                .stream()
                .map(this::toSessionResponse)
                .toList();
    }

    public List<ChatMessageResponse> getSessionMessages(UUID sessionId, UUID userId) {
        ChatSession session = sessionRepository.findByIdAndUserId(sessionId, userId)
                .orElseThrow(() -> new ResourceNotFoundException("ChatSession", "id", sessionId));

        return messageRepository.findBySessionIdOrderByCreatedAtAsc(session.getId())
                .stream()
                .map(this::toMessageResponse)
                .toList();
    }

    @Transactional
    public void deleteSession(UUID sessionId, UUID userId) {
        ChatSession session = sessionRepository.findByIdAndUserId(sessionId, userId)
                .orElseThrow(() -> new ResourceNotFoundException("ChatSession", "id", sessionId));
        sessionRepository.delete(session);
    }

    private ChatSessionResponse toSessionResponse(ChatSession session) {
        return ChatSessionResponse.builder()
                .id(session.getId())
                .documentId(session.getDocumentId())
                .title(session.getTitle())
                .messageCount(session.getMessages().size())
                .createdAt(session.getCreatedAt())
                .build();
    }

    private ChatMessageResponse toMessageResponse(ChatMessage message) {
        return ChatMessageResponse.builder()
                .id(message.getId())
                .role(message.getRole())
                .content(message.getContent())
                .sourceChunks(message.getSourceChunks())
                .createdAt(message.getCreatedAt())
                .build();
    }
}
